### PR TITLE
Update 3.5.0

### PR DESCRIPTION
**Pull request by vbuberen**
## :page_facing_up: Context
This PR is a set of changes required to release a new version of Chucker with support Android 12.
It has not all changes from `develop`, but all non-breaking ones and changes required to support Android 12.

## :pencil: Changes
- #593 PR for compatibility with Android 12
- All dependencies updates
- A few other non-breaking fixes

## :no_entry_sign: Breaking
Nothing breaking expected as I tried to take only parts we need.

## :hammer_and_wrench: How to test
No special testing required. Tried to run both on an emulator with API 19 and on one with Android S

## :stopwatch: Next steps
I will create a follow up PRs with updates to changelogs and readme files.

### CODE REVIEW

File: library/build.gradle [hunk 1]
```diff
@@ -23,11 +23,16 @@ android {
 
     buildFeatures {
         viewBinding = true
+        buildConfig = false
```
Comment (MiSikora) [inline library/build.gradle:26]: This is in theory a breaking change but I think we can make it slide. I don't suppose anyone uses `BuildConfig`.

**Review by MiSikora — APPROVED** [review]

File: library/build.gradle (hunk 1 above)
Comment (vbuberen) [reply library/build.gradle]: Good catch. I would like to play safe and be sure that we don't introduce anything breaking for 3.x users, so reverted this change.

File: gradle/gradle-mvn-push.gradle [hunk 2]
```diff
@@ -95,6 +95,10 @@ afterEvaluate {
                             name = 'Olivier Perez'
                             email = 'olivier@olivierperez.fr'
                         }
+                        developer {
```
Comment (vbuberen) [inline gradle/gradle-mvn-push.gradle:98]: @MiSikora Found out that we had you missing for quite some time here in POM. Decided to fix it here. Will do a PR for `develop` as well later.
Let me know if this data is fine (took it from one of your repos 😄 ) and if you would like to add your email as well here.

Comment (MiSikora) [reply gradle/gradle-mvn-push.gradle:98]: Yup, looks good. Skipping email is fine :).

**Pull request merged**